mne/stats/regression.py

```diff
@@ -256,7 +256,39 @@ def solver(X, Y):
         else:
             raise ValueError("No such solver: {0}".format(solver))
 
-    # prepare raw and events
+    # build data
+    data, info, events = _prepare_rerp_data(raw, events, picks=picks,
+                                            decim=decim)
+
+    # build predictors
+    X, conds, cond_length, tmin_s, tmax_s = _prepare_rerp_preds(
+        n_samples=data.shape[1], sfreq=info["sfreq"], events=events,
+        event_id=event_id, tmin=tmin, tmax=tmax, covariates=covariates)
+
+    # remove "empty" and contaminated data points
+    X, data = _clean_rerp_input(X, data, reject, flat, decim, info, tstep)
+
+    # solve linear system
+    coefs = solver(X, data)
```

Comment:
Let's see if this is worth a separate issue, should be systematic then across the package. Let's not touch it here.